include/alpaka/rand/Philox/PhiloxBaseTraits.hpp

```diff
@@ -0,0 +1,80 @@
+/* Copyright 2021 Jiri Vyskocil
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <alpaka/rand/Philox/PhiloxBaseCommon.hpp>
+#include <alpaka/rand/Philox/PhiloxBaseStdArray.hpp>
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED) || defined(ALPAKA_ACC_GPU_HIP_ENABLED)
+#    include <alpaka/rand/Philox/PhiloxBaseCudaArray.hpp>
+#endif
+
+#include <alpaka/alpaka.hpp>
+
+#pragma once
+
+
+namespace alpaka
+{
+    namespace rand
+    {
+        namespace engine
+        {
+            namespace traits
+            {
+                /** Selection of default backend
+                 *
+                 * Selects the data backend based on the accelerator device type. As of now, different backends operate
+                 * on different array types.
+                 *
+                 * @tparam TAcc the accelerator as defined in alpaka/acc
+                 * @tparam TParams Philox algorithm parameters
+                 * @tparam TImpl engine type implementation (CRTP)
+                 * @tparam TSfinae internal parameter to stop substitution search and provide the default
+                 */
+                template<typename TAcc, typename TParams, typename TImpl, typename TSfinae = void>
+                struct PhiloxBaseTraits
+                {
+                    using Backend = PhiloxBaseStdArray<TParams, TImpl>; ///< Selected device-dependent backend
+                                                                        ///< implementation: std::array
+
+                    using Counter = typename Backend::Counter; ///< Counter array type
+                    using Key = typename Backend::Key; ///< Key array type
+
+                    using Base = PhiloxBaseCommon<Backend, TParams, TImpl>; ///< Base type to be inherited from
+                };
+
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED) || defined(ALPAKA_ACC_GPU_HIP_ENABLED)
+                /** CUDA backend specialization
+                 *
+                 * @tparam TAcc the accelerator as defined in alpaka/acc
+                 * @tparam TParams Philox algorithm parameters
+                 * @tparam TImpl engine type implementation (CRTP)
+                 */
+                template<typename TAcc, typename TParams, typename TImpl>
+                struct PhiloxBaseTraits<
+                    TAcc,
+                    TParams,
+                    TImpl,
+                    typename std::enable_if<
+                        std::is_same_v<typename alpaka::traits::DevType<TAcc>::type, alpaka::DevUniformCudaHipRt>>>
```

Comment:
Instead of using `enable_if`, can you specialize on `TAcc` directly? Something like:
```c++
template<...>
struct PhiloxBaseTraits<alpaka::AccUniformCudaHipRt<...>, ...> { ... };
```